gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerFragment.java

```diff
@@ -163,6 +146,24 @@ private void onMapReady(MapAdapter map) {
     binding.moveFeature.cancelButton.setOnClickListener(__ -> setDefaultMode());
     enableLocationLockBtn();
     mapContainerViewModel.getMbtilesFilePaths().observe(this, map::addTileOverlays);
+    mapContainerViewModel.getSelectMapTypeClicks()
+        .as(autoDisposable(getViewLifecycleOwner()))
```

Comment:
I also considered using a flowable processor which converts events to LiveData. But the problem with that is, it always emits the last value on resubscription. So, the dialog would pop-up again and again after every rotation (or any config change).

Reply:
That shouldn't be the case; Flowables would only emit the last item if you use buffer strategy "latest". I think this is the most likely cause, since `selectMapTypeClicks` is a `PublishSubject`, which doesn't cache the last value.

If you're still having trouble with duplicate requests, consider using `com.google.android.gnd.rx.Event` to wrap the value, and only processing it using its `ifUnhandled()` method.

Reply:
Done